nemo/collections/asr/parts/utils/streaming_utils.py

```diff
@@ -1346,7 +1351,10 @@ def __iter__(self):
                 )
 
             if self.buffer_idx == 0 and isinstance(self.streaming_cfg.shift_size, list):
-                shift_size = self.streaming_cfg.shift_size[0]
+                if self.pad_and_drop_preencoded:
```

Comment:
How about adding the self.pad_and_drop_preencoded condition to the main if block if makes it easier to read?